### DISCUSSION TITLE
Custom Search Paths

### DISCUSSION THREAD

**Discussion by jurialmunkey** [jurialmunkey/skin.arctic.horizon.2#1014] [Discussion]
Transferred from AH1/AZ2

This issue is a place for people to share their custom search paths. You can input custom paths via
`Skin Settings > Menus > Customise search menu > Action > Custom Item >`

The search paths append the search string to the end of the path. You must organise the path so that where you pass the search query will be at the end.

For instance, say I search for the movie Alien and my Container.FolderPath for that query looks like this:
`plugin://plugin.video.mycoolplugin/?info=search&query=Alien&type=movie`

To make it work with the search window your path would need to be rearranged like below so that the search term being appended on the end will line up properly with the query= part:
`plugin://plugin.video.mycoolplugin/?info=search&type=movie&query=`

**Comment by jurialmunkey** [comment]
### Twitch
**Streams**
`plugin://plugin.video.twitch/?content=streams&mode=search_results&index=0&query=`

**Channels**
`plugin://plugin.video.twitch/?content=channels&mode=search_results&index=0&query=`

**Games**
`plugin://plugin.video.twitch/?content=games&mode=search_results&index=0&query=`

**Comment by jurialmunkey** [comment]
### Spotify
**Artists**
`plugin://plugin.audio.spotify/?action=search_artists&artistid=`

**Albums**
`plugin://plugin.audio.spotify/?action=search_albums&albumid=`

**Songs**
`plugin://plugin.audio.spotify/?action=search_songs&songid=`

**Playlists**
`plugin://plugin.audio.spotify/?action=search_playlists&playlistid=`

**Comment by iali98** [comment]
### Seren 1.0 and above

**Movies**
`plugin://plugin.video.seren/?action=moviesSearchResults&actionArgs=`

**Shows**
`plugin://plugin.video.seren/?action=showsSearchResults&actionArgs=`

**Comment by rapteev** [comment]
Hi
Thank you for this feature.
What about search local library? Is it possible with this feature? 
I am using emby for kodi for my libraries and I do not know how to implement this feature on my library.
Thank you

**Comment by chiefkes** [comment]
Openmeta possible?

**Comment by jurialmunkey** [comment]
> 
> 
> Hi
> Thank you for this feature.
> What about search local library? Is it possible with this feature?
> I am using emby for kodi for my libraries and I do not know how to implement this feature on my library.
> Thank you

It searches local library by default. You can access the search function from the top menu in the library or from the search icon on the horizontal fixed width home menu.

**Comment by drinfernoo** [comment]
I have a question on how this is implemented at skin-level: does the skin set a `Skin.String(...)` value for the search term and then just append that on the end of the URI? There are some add-ons that have the search term somewhere in the middle, but I would think inserting a `$INFO[Skin.String(...)]` into the URI would make it work?

It appears that it's defined in `Includes_Object.xml`:
https://github.com/jurialmunkey/skin.arctic.zephyr.2/blob/c793a6a87b3a10d4383c29479e5bb69fa1313729/1080i/Includes_Object.xml#L26

But how can we manipulate that for non-standard paths?

**Comment by jurialmunkey** [comment]
@drinfernoo - Whatever path you define as the search action gets put into $PARAM[path] and then the skin.string gets tacked onto that as you can see in the `<content>` tags you mention above.

Some of the built-in library searches need the search string in the middle, so that's why there is a path2 param. I should be able to incorporate this end line into the setup dialog - though it would have to be a completely manual entry of the path ending.

However, I haven't come across an add-on where the search term has to be in the middle. That's because in plugin paths you should be able to re-organise any arguments separated by an ampersand (&) so that the search query is at the end. 

For instance say this is your search path:
`plugin://plugin.video.twitch/?content=streams&mode=search_results&query=SEARCHTERM&index=0`

You can re-arrange that so that the search query is at the end and it will still work:
`plugin://plugin.video.twitch/?content=streams&mode=search_results&index=0&query=SEARCHTERM`

**Comment by drinfernoo** [comment]
I know of one specifically that was built with `xbmcswift`, which makes it's paths a lot different.

Would it be possible we can simply include the InfoLabel in our path, and the XML would know what to do with it?

**Comment by algoldbe** [comment]
I have a string for Elementum Movies and another one for TV Shows. My searches with one word work fine, but if I add a space, there's and error about bad request. If I use WORD+WORD it works fine. How can I use spaces instead of "+" in the queries? Thanks!

**Comment by drinfernoo** [comment]
@algoldbe That is probably an issue with the way Elementum parses search strings. You'd need to take it up with the maintainer of that add-on.

**Comment by thedefside** [comment]
**Internet Archive Game Launcher**

**All Lists**
`plugin://plugin.program.iagl/run_search/1/?title=`

**Atari 2600, NES, SNES, N64, Sega Genesis, PS1, MAME 2003+**
`plugin://plugin.program.iagl/run_search/1/?genre=None&lists=%5bu%27Atari_2600_ZachMorris%27%2c%20u%27Genesis_ZachMorris%27%2c%20u%27MAME_2003_Plus_ZachMorris%27%2c%20u%27N64_ZachMorris%27%2c%20u%27NES_ZachMorris%27%2c%20u%27PS1_ZachMorris%27%2c%20u%27SNES_ZachMorris%27%5d&nplayers=None&studio=None&tag=None&year=None&title=`

**Comment by notoco** [comment]
I noticed a problem.
After searching for a movie and selecting it for playback from the widget, before it begins to play, a busy window appears, and under it you can see that the widgets are refreshing.
This is not visible with one search engine, but e.g. when there are 5 of them and a movie from the 4-5 widget is selected, you can see the background refreshing of the widgets with search results. This slows down kodi a little (coreelec 9.2 on s905x2) and makes unnecessary network traffic in my opinion.

**Comment by jurialmunkey** [comment]
> 
> 
> I noticed a problem.
> After searching for a movie and selecting it for playback from the widget, before it begins to play, a busy window appears, and under it you can see that the widgets are refreshing.
> This is not visible with one search engine, but e.g. when there are 5 of them and a movie from the 4-5 widget is selected, you can see the background refreshing of the widgets with search results. This slows down kodi a little (coreelec 9.2 on s905x2) and makes unnecessary network traffic in my opinion.

Nothing I can do about that. Kodi core determines when containers update, not the skin (there are some tricks the skin can do to force a widget refresh but nothing can be done to prevent one).

**Comment by 7E1EVEN** [comment]
Hi, I can’t seem to figure out the search paths for TMDbHelper. 
Can someone help me out?

Thanks.

**Comment by drinfernoo** [comment]
@7E1EVEN They are included by default in the skin already.

**Comment by 7E1EVEN** [comment]
@drinfernoo I don’t seem to have them.

**Comment by drinfernoo** [comment]
`Settings -> Skin Settings -> Miscellaneous -> Choose search lists`

They're in there by default, on any of the entries, choose "Default search items", and you'll find them at the bottom.

**Comment by 7E1EVEN** [comment]
@drinfernoo aah, thanks. 
Never came up to me to look there.

**Comment by sayrnla** [comment]
is it possible to get WonderfulSubs added? I've tried a few times to no avail.

**Comment by DaisyF8** [comment]
WonderfulSubs :

plugin://plugin.video.wonderfulsubs/search/

but I don't know if this answer qualifies since the results produced aren't links to videofiles, each answer-instance instead links to a directory. That's also why I can't be sure of the search-string above, because I haven't seen any results.

**Comment by DaisyF8** [comment]
Foxystreams

plugin://plugin.video.foxystreams/?mode=search&query=

Would need some default art I think.

**Comment by DaisyF8** [comment]
I notice that when I do a search and scroll in the widgets through the results, if I do a new search the position in the widgets doesn't get reset to their start position (totally to the left). So for an affected widget in the new search results you need to scroll left to see the first result.

**Comment by popeye2468** [comment]
any chance we can get watchnixtoons2 and composite for plex

**Comment by thedefside** [comment]
> 
> 
> any chance we can get watchnixtoons2 and composite for plex

@popeye2468 

### WatchNixtoons2

**Series:**
`plugin://plugin.video.watchnixtoons2/?action=actionCatalogSection&section=ALL&searchType=series&path=%2Fsearch&query=`
**Episodes:**
`plugin://plugin.video.watchnixtoons2/?action=actionCatalogSection&section=ALL&searchType=episodes&path=%2Fsearch&query=`
**Movies:**
`plugin://plugin.video.watchnixtoons2/?action=actionCatalogSection&section=ALL&searchType=movies&path=%2Fsearch&query=`

**Comment by dleast** [comment]
> ### Spotify
> **Artists**
> `plugin://plugin.audio.spotify/?action=search_artists&artistid=`
> 
> **Albums**
> `plugin://plugin.audio.spotify/?action=search_albums&albumid=`
> 
> **Songs**
> `plugin://plugin.audio.spotify/?action=search_songs&songid=`
> 
> **Playlists**
> `plugin://plugin.audio.spotify/?action=search_playlists&playlistid=`

The path for searching songs/tracks has been changed:
plugin://plugin.audio.spotify/?action=search_tracks&trackid=

**Comment by dleast** [comment]
Did anyone succeed in adding Netflix plugin search results? I tried several paths:

plugin://plugin.video.netflix/?find=
plugin://plugin.video.netflix/?profile_guid=XXXXX&find=
plugin://plugin.video.netflix/?switch_profile_guid=XXXXX&find=
plugin://plugin.video.netflix/?autoselect_profile_guid=True&find=

Unfortunately all of them return Netflix Profiles as result.

**Comment by drinfernoo** [comment]
@dleast Try using the paths that are in the TMDbHelper Netflix player file, under `search_movie` and `search_episode` 😉

**Comment by Agraphie** [comment]
For Amazon VOD I used the following. Unfortunately, the result also returns non-bought/non-prime videos, but in general it works.

```
plugin://plugin.video.amazon-test/?mode=Search&searchstring=
```

**Comment by Takezo36** [comment]
Not mine but joyn uses this
plugin://plugin.video.joyn/?search_term=<search_item>&mode=search

**Comment by starman2908** [comment]
> any chance we can get watchnixtoons2 and composite for plex @popeye2468 

for plex kodi connect...I got mine by adding the search function within each plex node from inside the addon to my favourites (Movies, TV Shows etc). Then opened my favourites in a text viewer... should look like this...

favourite name="Search Shows..." thumb="special://home/addons/plugin.video.plexkodiconnect/icon.png">ActivateWindow(10025,&quot;plugin://plugin.video.plexkodiconnect/?id=40&amp;key=%2flibrary%2fsections%2f40%2fsearch%3ftype%3d2&amp;mode=browseplex&amp;prompt=Search%20for%20Shows&quot;,return)</favourite

An example of the bit you need for the custom search list is in bold below but substitute **&quot** for **&query=** so it looks like this..

**plugin://plugin.video.plexkodiconnect/?id=40&amp;key=%2flibrary%2fsections%2f40%2fsearch%3ftype%3d2&amp;mode=browseplex&amp;prompt=Search%20for%20Shows&query=**

I'm not great at code so hope that makes sense. The above example is how it appears in mine, your nodes within the addon will have different order and content to my setup, but i got it working by doing the above. 

TIP Remember the order you add your searches from Plex Kodi Connect to your favourites, i just found it easier to name them when setting up the custom search.

**Comment by alKODIque** [comment]
For vStream, french movies & tv shows addon:

MOVIES:
plugin://plugin.video.vstream/?function=showSearch&sCat=1&site=globalSearch&siteUrl=http%3a%2f%2fvenom&title=Recherche%20(Films)&action=activate&searchtext=

TV SHOWS:

plugin://plugin.video.vstream/?function=showSearch&sCat=2&site=globalSearch&siteUrl=http%3a%2f%2fvenom&title=Recherche%20(S%c3%a9ries%2fAnim%c3%a9s)&action=activate&searchtext=

**Comment by ChristofBuch** [comment]
I know this has beed asked before but I cannot find the Netflix search path.
Also I try to find it in TMDBhelper but I dont know how this could work.
Does anyone know what the path is?

**Comment by Drakses** [comment]
Hi! I have a strange issue with my custom search. I'm using `&actionArgs=` at the end. For some reason after search input it begins to search and after a short time the search window appears for the second time. If I input the search term again, it finally return the results. Can someone help me please 😁

**Comment by drinfernoo** [comment]
@Drakses It will all depend on the add-on and whether or not you're using the correct path.

**Comment by Drakses** [comment]
Hmm...I am using the correct path, after following the author instructions. So the plug-in remains....I'm wondering if I can do something about it.

**Comment by vermope** [comment]
For Belgium TV Channels:

VRT NU: 
plugin://plugin.video.vrt.nu/search/query/

VTM GO:
plugin://plugin.video.vtm.go/search/

VIER-VIJF-ZES:
plugin://plugin.video.viervijfzes/search/

**Comment by mattipunkt** [comment]
How's the path for Netflix now?

**Comment by jurialmunkey** [comment]
> 
> 
> How's the path for Netflix now?

Sadly it doesn't work any more because the Netflix plugin stopped supporting passing a search query directly to the plugin path.

**Comment by Bulletbling** [comment]
Someone provided the search code for Seren and it seems it is implemented into the skin now. Thanks!

what's weird is nobody has found the search code for the most popular add-ons, such as venom, the crew, Exodus redux, fen, etc.

If I knew how to find these or you can tell me, I would do it!

**Comment by starmanrf** [comment]
@Bulletbling for venom, this worked for me:

Movies
plugin://plugin.video.venom/','12','?action=movieSearchterm&name=

TV Shows:
plugin://plugin.video.venom/?action=tvSearchterm&name=

To discover more, I would suggest run Kodi, open an add-on and run a search for a Film/TV show. You could then quit Kodi and find in your debug log the item you searched for. Then It should be just a case  of copying the code from there, minus the search term.

**Comment by Bulletbling** [comment]
> @Bulletbling for venom, this worked for me:
> 
> Movies
> plugin://plugin.video.venom/','12','?action=movieSearchterm&name=
> 
> TV Shows:
> plugin://plugin.video.venom/?action=tvSearchterm&name=
> 
> To discover more, I would suggest run Kodi, open an add-on and run a search for a Film/TV show. You could then quit Kodi and find in your debug log the item you searched for. Then It should be just a case of copying the code from there, minus the search term.

Worked exactly as you wrote it. Thanks!

**Comment by AakashC2020** [comment]
Hi,
Could someone provide the custom search paths (movies & tv shows) for the "Fen" add-on?
Thanks in advance!

**Comment by AakashC2020** [comment]
Custom search paths for the Fen add-on as provided by its developer Tikipeter:

MOVIES:
plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=

EPISODES:
plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=

**Comment by Bulletbling** [comment]
> Custom search paths for the Fen add-on as provided by its developer Tikipeter:
> 
> MOVIES:
> plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=
> 
> EPISODES:
> plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=

I wasn't the one to ask for this, but thank you! Will help others as well. There used to not be an easy way to search all video add-ons at once in kodi. Glad we have made progress.

**Comment by AakashC2020** [comment]
> > Custom search paths for the Fen add-on as provided by its developer Tikipeter:
> > MOVIES:
> > plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=
> > EPISODES:
> > plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=
> 
> I wasn't the one to ask for this, but thank you! Will help others as well. There used to not be an easy way to search all video add-ons at once in kodi. Glad we have made progress.

You're welcome! :)

**Comment by jamesvu98** [comment]
Does anyone happen to know the search path for kaito?

**Comment by Goldenfreddy0703** [comment]
Yeah i also wanna know the search path too for Kaito.

**Comment by jamesvu98** [comment]
I somewhat figured it out:

plugin://plugin.video.kaito/search/

But you have to append /1 to the search. E.g. if I want to search for attack on titan I'd type "attack on titan/1"

**Comment by Goldenfreddy0703** [comment]
Thank you so much for helping me on that, it really helped my anime section but i believe there is a way to git rid of the one.
Maybe later i can talk with the developer and if he doesn't know what to do on that, maybe i can modify to addon.xml and make the search normal like the Honeybird kodi adoon sense there was a way to add a custom search for it. 
Again thanks for the support and this really helped me out.

Btw, here is the search configuration on how the hummingbird add-on did it:`ActivateWindow(Videos,”plugin://plugin.video.hummingbird/?action=title_search&page=1&title=$INFO[Skin.String(CustomSearchTerm)]”,return)`

https://github.com/Wilson-Magic/plugin.video.hummingbird/issues/17

**Comment by popeye2468** [comment]
does anyone know how to add tidal2 addon?

**Comment by xaviron** [comment]
you can use it also with DISNEY+, it works great, I've just discovered it by essay and error

`plugin://slyguy.disney.plus/?_search&query=`

or

`activatewindow(videos,"plugin://slyguy.disney.plus/?_search&query=",return)`

Does anyone know how to make it work for:

- Alfa
- Balandro
- Cristal Azul
- Palantir2
- Netflix
- HBO

?? Thank you!

**Comment by xaviron** [comment]
I've got a solution for NETFLIX... but you have to change the plugin code, and it works perfectly.

The search command should be:
`plugin://plugin.video.netflix/directory/search/search/add2/?textobus=`

and to get it working follow this steps:

1) Make a backup of the file **_directory_search.py_** located at **_plugin.video.netflix\resources\lib\navigation_** for security reasons.
2) Now edit the file, look for _**def route_search_nav**_ and locate the code:
`elif path == 'add':
        ret = search_add()`
and following it ADD this code:
`elif path == 'add2':
        ret = search_add2(params['textobus'])`

3) Look for the function _**def search_add()**_ and leave it untouched, and following it ADD this other new function:


```
def search_add2(textobus): 

    row_id = G.LOCAL_DB.insert_search_item('text', textobus)

    if row_id is not None and search_query(row_id, 0, False):

        url = common.build_url(['search', 'search', row_id], mode=G.MODE_DIRECTORY, params={'dir_update_listing': True})

        common.container_update(url, False)

        return True

    return False
```

4) Now it will work perfectly. 

NOTE: Remember that if you update the plugin this code can be replaced, so you'll have to do it again, just copy and paste over the new one.

**Comment by CastagnaIT** [comment]
FYI i have added to Netflix add-on this endpoint to allow a text search without needed GUI interactions:
`plugin://plugin.video.netflix/directory/search/search/?query=something`

this change is in master branch will be included on next release

**Comment by mattipunkt** [comment]
> you can use it also with DISNEY+, it works great, I've just discovered it by essay and error
> 
> `plugin://slyguy.disney.plus/?_search&query=`
> 
> or
> 
> `activatewindow(videos,"plugin://slyguy.disney.plus/?_search&query=",return)`

This does not work for me. Anyone else facing this issue?

**Comment by Goldenfreddy0703** [comment]
Hey guys, if anyone is testing the new beta of Kaito 0.0.11, here is the new custom search path for Kaito 0.0.11.

`plugin://plugin.video.kaito/?action=search_results&action_args=`

Big thanks to SwagOtaku for providing this.

**Comment by Liquidrag** [comment]
> Hey guys, if anyone is testing the new beta of Kaito 0.0.11, here is the new custom search path for Kaito 0.0.11.
> 
> `plugin://plugin.video.kaito/?action=search_results&action_args=`
> 
> Big thanks to SwagOtaku for providing this.

Thank you and SwagOtaku for this!

**Comment by rapteev** [comment]
Hi
Anyone have any luck with composite plugin?
I tried a lot but nothing works.
Thank you

**Comment by Liquidrag** [comment]
REQUEST: Does anyone have the paths for MP3 Streams?

**Comment by krzysztofwinnik** [comment]
For HBO Max integrated search results (TV Series + Movies) use:
`plugin://slyguy.hbo.max/?_=search&amp;query=`

**Comment by patkino** [comment]
Elementum 
Movies 
`plugin://plugin.video.elementum/movies/search?q=`
Shows
`plugin://plugin.video.elementum/shows/search?q=`
Works only with ascii coding. If anyone knows how to modify string to use Cyrillic- please let know

**Comment by meesterexx** [comment]
Crunchyroll: https://github.com/MrKrabat/plugin.video.crunchyroll
plugin://plugin.video.crunchyroll/?mode=search&args=

This works, but only after it pops up an empty onscreen keyboard. If you hit enter in the AH2 search bar to bring up the os keyboard first before entering any text, it will still popup an empty os keyboard after entering search text and starting the search, forcing you to enter the search text a second time.

So it's not accepting the initial search arg for Crunchyroll, it only accepts search arg from when the crunchyroll python script prompts for it.

I'm not a dev so I don't know if this is a limitation of how the crunchyroll python executes a search, or if I don't have the custom search path string correct.

Can someone advise or get it working? 
The crunchyroll.py does a check_mode and if mode == "search" it calls searchAnime(args) in the controller.py
(Hopefully someone better versed in python can look at this and see if it can work for a search path without the extra empty os keyboard popup)

**Comment by rapteev** [comment]
Hi
I only miss Tidal and Composite plugins.
Anyone knows the search paths for these?
Thank you

**Comment by ray2301** [comment]
> Hi I only miss Tidal and Composite plugins. Anyone knows the search paths for these? Thank you

for tidal, yes. i'm waiting for arneson to implement it :)
read from here:
https://forum.kodi.tv/showthread.php?tid=200555&pid=3102859#pid3102859
edit: oh yes, you already know :D

**Comment by Mr-7mdan** [comment]
> For vStream, french movies & tv shows addon:
> 
> MOVIES: plugin://plugin.video.vstream/?function=showSearch&sCat=1&site=globalSearch&siteUrl=http%3a%2f%2fvenom&title=Recherche%20(Films)&action=activate&searchtext=
> 
> TV SHOWS:
> 
> plugin://plugin.video.vstream/?function=showSearch&sCat=2&site=globalSearch&siteUrl=http%3a%2f%2fvenom&title=Recherche%20(S%c3%a9ries%2fAnim%c3%a9s)&action=activate&searchtext=

Did you manage to search a certain website/source? 
when I tried, the path created doesn't seem to have search term at all. So, I tried everything around it but didn't work. 
@alKODIque

**Comment by Mr-7mdan** [comment]
For MDBList inside TMDB Helper

plugin://plugin.video.themoviedb.helper/?info=mdblist_searchlists&tmdb_type=both&widget=true&query=

**Comment by Mr-7mdan** [comment]
For TMDB Helpper Keywords search

plugin://plugin.video.themoviedb.helper/?info=search&tmdb_type=keyword&widget=true&query=

**Comment by FieryMewtwo** [comment]
Anyone know the strings for YouTube?

**Comment by ray2301** [comment]
> Anyone know the strings for YouTube?

the one for youtube has two lines (this is from template.xml):
```
            <property name="widgetPath" tag="property" attribute="name|list" value="DefaultSearch-Youtube">plugin://plugin.video.youtube/search/?q=</property>
            <property name="widgetPath2" tag="property" attribute="name|list" value="DefaultSearch-Youtube">&amp;search_type=notvalid</property>
```
so the full string is `plugin://plugin.video.youtube/search/?q="YOUR SEARCH TERM"&amp;search_type=notvalid`
but youtube search is already implemented and selectable within the skin so you don't really need this since that works.

**Comment by FieryMewtwo** [comment]
oh. okay

**Comment by Goldenfreddy0703** [comment]
## Otaku
Hey guys, me and my team just released a new anime addon called Otaku. If anybody would like the custom search paths for the skin.
Here you go. 

### Both
`plugin://plugin.video.otaku/search/both/`

### Movies
`plugin://plugin.video.otaku/search/movie/`

### TV Shows
`plugin://plugin.video.otaku/search/show/`

**Comment by ren2r** [comment]
> Crunchyroll: https://github.com/MrKrabat/plugin.video.crunchyroll plugin://plugin.video.crunchyroll/?mode=search&args=
> 
> This works, but only after it pops up an empty onscreen keyboard. If you hit enter in the AH2 search bar to bring up the os keyboard first before entering any text, it will still popup an empty os keyboard after entering search text and starting the search, forcing you to enter the search text a second time.
> 
> So it's not accepting the initial search arg for Crunchyroll, it only accepts search arg from when the crunchyroll python script prompts for it.
> 
> I'm not a dev so I don't know if this is a limitation of how the crunchyroll python executes a search, or if I don't have the custom search path string correct.
> 
> Can someone advise or get it working? The crunchyroll.py does a check_mode and if mode == "search" it calls searchAnime(args) in the controller.py (Hopefully someone better versed in python can look at this and see if it can work for a search path without the extra empty os keyboard popup)

For Crunchyroll
`plugin://plugin.video.crunchyroll/?mode=search&search=`

**Comment by meesterexx** [comment]
> For Crunchyroll `plugin://plugin.video.crunchyroll/?mode=search&search=`

That worked great, thanks!

**Comment by joseska** [comment]
> you can use it also with DISNEY+, it works great, I've just discovered it by essay and error
> 
> `plugin://slyguy.disney.plus/?_search&query=`
> 
> or
> 
> `activatewindow(videos,"plugin://slyguy.disney.plus/?_search&query=",return)`
> 
> Does anyone know how to make it work for:
> 
>     * Alfa
> 
>     * Balandro
> 
>     * Cristal Azul
> 
>     * Palantir2
> 
>     * Netflix
> 
>     * HBO
> 
> 
> ?? Thank you!

+1

**Comment by FieryMewtwo** [comment]
Netflix is included I think. HBO Max uses the same as the other SlyGuy plugins, just change it to reference the HBO Max plugin. As to the others there, I don't use them, so I wouldn't be able to provide any information about them.

**Comment by Goldenfreddy0703** [comment]
* Netflix
`plugin://plugin.video.netflix/directory/search/search/?query=`
* HBO
`plugin://slyguy.hbo.max/?_=search&amp;query=`

Same here, I don't use them, either.

**Comment by umbrellaplug** [comment]
Umbrella

- TV
`plugin://plugin.video.umbrella/?action=tvSearchterm&name=`
-Movies
`plugin://plugin.video.umbrella/?action=movieSearchterm&name=`

**Comment by GrimmInsomnia** [comment]
Thank you all for the custom search paths provided!
I apologize if this may be slightly "off-topic", but:

I have been working on a custom skin and would like to use custom search paths of some Kodi 19 addons if anyone happens to know the paths or can work out them out more efficient than I can!

- Magic Dragon (Narcacist)*
- KodiVerse (Narcacist)*
- The Crew (TeamCrew)
- Exodus/Exodus Redux (I-A-C)

Thanks!

Updates:
- Magic Dragon (Narcacist) | `plugin://plugin.video.magicdragon/?mode=5`

**Comment by ambulancePilot** [comment]
Is anyone able to provide the search path for People and Keywords (both movie and TV show) in Fen? Thank you in advance!

**Comment by bassemhelal18** [comment]
> > For vStream, french movies & tv shows addon:
> 
> > 
> 
> > MOVIES: plugin://plugin.video.vstream/?function=showSearch&sCat=1&site=globalSearch&siteUrl=http%3a%2f%2fvenom&title=Recherche%20(Films)&action=activate&searchtext=
> 
> > 
> 
> > TV SHOWS:
> 
> > 
> 
> > plugin://plugin.video.vstream/?function=showSearch&sCat=2&site=globalSearch&siteUrl=http%3a%2f%2fvenom&title=Recherche%20(S%c3%a9ries%2fAnim%c3%a9s)&action=activate&searchtext=
> 
> 
> 
> Did you manage to search a certain website/source? 
> 
> when I tried, the path created doesn't seem to have search term at all. So, I tried everything around it but didn't work. 
> 
> @Mr-7mdan 
> 
> did you find a way ?

**Comment by caisamuels** [comment]
Anyone got search paths for Scrubs V2?

**Comment by MrC001** [comment]
Scrubs V2

ScrubsV2 - Movies
plugin://plugin.video.scrubsv2/?action=movies_searchterm&select=movies&name=

ScrubsV2 - TV Shows
plugin://plugin.video.scrubsv2/?action=tvshows_searchterm&select=tvshow&name=

**Comment by Sugar-devil** [comment]
As always @jurialmunkey ,ah2 is incredible, huge props..
For these searches,do I edit the code or this is just to add search through the main menu?
Also do I use these same strings for Titan Bingie?
Thank u, in advance

**Comment by ashbruh** [comment]
Fen

Fen - Movies
plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=

Fen - TV Shows
plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=

**Comment by matthius7** [comment]
> Fen
> 
> Fen - Movies plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=
> 
> Fen - TV Shows plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=

Do you have search query for Fen or tmdb for Collections? I tried the one on Reddit and it's coming up blank. And I deleted the tmdb collection thinking this was why Fen collections that I added wouldn't show. I can restore defaults but will have to set back up widgets for home page as well and that is much work. Please and thank you beforehand.

**Comment by Omarscomin** [comment]
> > Fen
> 
> > 
> 
> > Fen - Movies plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=
> 
> > 
> 
> > Fen - TV Shows plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=
> 
> 
> 
> Do you have search query for Fen or tmdb for Collections? I tried the one on Reddit and it's coming up blank. And I deleted the tmdb collection thinking this was why Fen collections that I added wouldn't show. I can restore defaults but will have to set back up widgets for home page as well and that is much work. Please and thank you beforehand. 

@matthius7 

Kodi settings > Skin > Menus > Customise search menu > Highlight any widget > Restore menu items > Reset to skin defaults

That will restore your search menu widgets **only**.

**Comment by matthius7** [comment]
> > > Fen
> > 
> > 
> > > 
> > 
> > 
> > > Fen - Movies plugin://plugin.video.fen/?mode=build_movie_list&action=tmdb_movies_search&query=
> > 
> > 
> > > 
> > 
> > 
> > > Fen - TV Shows plugin://plugin.video.fen/?mode=build_tvshow_list&action=tmdb_tv_search&query=
> > 
> > 
> > Do you have search query for Fen or tmdb for Collections? I tried the one on Reddit and it's coming up blank. And I deleted the tmdb collection thinking this was why Fen collections that I added wouldn't show. I can restore defaults but will have to set back up widgets for home page as well and that is much work. Please and thank you beforehand.
> 
> @matthius7
> 
> Kodi settings > Skin > Menus > Customise search menu > Highlight any widget > Restore menu items > Reset to skin defaults
> 
> That will restore your search menu widgets **only**.

Very nice thank you.

**Comment by Bi0nicAztec** [comment]
Umbrella paths?

**Comment by Omarscomin** [comment]
> Umbrella paths?

@Bi0nicAztec 

Movies: 
plugin://plugin.video.umbrella/?action=movieSearchterm&name= 

TV shows: 
plugin://plugin.video.umbrella/?action=tvSearchterm&name= 

Trakt Lists (Movies): 
plugin://plugin.video.umbrella/?action=trakt_search_lists&media_type=movies&name=

Trakt Lists (TV Shows):
plugin://plugin.video.umbrella/?action=trakt_search_lists&media_type=tvshows&name=

**Comment by rapteev** [comment]
Hi
Anyone knows search paths for mp3streams?
Thank you very much.

**Comment by juanjosegura** [comment]
How can I exctract or know the search path of an addon? It's called Palantir3 but I can't find how to do it

**Comment by Liquidrag** [comment]
Anyone have the search path for DramaCool?

**Comment by nomoxd** [comment]
Anyone have the search path for Otaku?

**Reply by Liquidrag** [reply]
https://github.com/jurialmunkey/skin.arctic.horizon.2/discussions/1014#discussioncomment-8424602
